src/integrations/Criteo/utils.js

```diff
@@ -87,46 +94,114 @@ const handleProductView = (message, finalPayload) => {
+  return false;
+};
+
+/**
+ * Returns transformed products array
+ * @param {*} properties
+ * @returns
+ */
+const getProductInfo = (properties) => {
   const productInfo = [];
-  let elementaryProduct;
-  if (properties && properties.products && properties.products.length > 0) {
+
+  if (properties?.products && properties.products.length > 0) {
     properties.products.forEach((product, index) => {
-      if (product.product_id && product.price && product.quantity) {
-        elementaryProduct = {
+      if (validateProduct(product, index)) {
+        productInfo.push({
           id: String(product.product_id),
           price: parseFloat(product.price),
           quantity: parseInt(product.quantity, 10),
-        };
-        if (
-          !Number.isNaN(parseFloat(elementaryProduct.price)) &&
-          !Number.isNaN(parseInt(elementaryProduct.quantity, 10))
-        ) {
-          // all the above fields are mandatory
-          productInfo.push(elementaryProduct);
-        }
-      } else {
-        logger.debug(`[Criteo] product at index ${index} is skipped for insufficient information`);
+        });
       }
     });
-    if (productInfo.length === 0) {
-      logger.debug(
-        '[Criteo] None of the products had sufficient information or information is wrongly formatted',
-      );
-      return;
-    }
   } else {
     logger.debug('[Criteo] Payload should consist of at least one product information');
+  }
+
+  return productInfo;
+};
+
+/**
+ * Adds order completed event to finalPayload
+ * @param {*} properties
+ * @param {*} finalPayload
+ * @param {*} productInfo
+ * @returns
+ */
+const processCompletedOrderEvent = (properties, finalPayload, productInfo) => {
+  const trackTransactionObject = {
+    event: 'trackTransaction',
+    id: String(properties.order_id),
+    item: productInfo,
+  };
+
+  if (!trackTransactionObject.id) {
+    logger.debug('[Criteo] order_id (Transaction Id) is a mandatory field');
+    return;
+  }
+
+  if (properties.new_customer === 1 || properties.new_customer === 0) {
+    trackTransactionObject.new_customer = properties.new_customer;
+  }
+
+  if (properties.deduplication === 1 || properties.deduplication === 0) {
+    trackTransactionObject.deduplication = properties.deduplication;
+  }
+
+  finalPayload.push(trackTransactionObject);
+};
+
+/**
+ * Adds view cart event to finalPayload
+ * @param {*} properties
+ * @param {*} finalPayload
+ * @param {*} productInfo
+ */
+const processViewedCartEvent = (finalPayload, productInfo) => {
+  const viewBasketObject = {
+    event: 'viewBasket',
+    item: productInfo,
+  };
+  finalPayload.push(viewBasketObject);
+};
+
+/**
+ * Handles events
+ * @param {*} message
+ * @param {*} finalPayload
+ * @returns
+ */
+const handlingEventDuo = (message, finalPayload) => {
+  const { event, properties } = message;
+  const eventType = event.toLowerCase().trim();
+  const productInfo = getProductInfo(properties);
+
+  if (productInfo.length === 0) {
```

Comment:
If there is an error logged from `getProductInfo()` we are again logging an error which is irrelevant